featuretools/tests/entityset_tests/test_ww_es.py

```diff
@@ -146,23 +152,79 @@ def test_init_es_with_relationships(pd_df):
         'dates_backwards': ['2020-09-09', '2020-09-08', '2020-09-07', '2020-09-06', '2020-09-05', '2020-09-04', '2020-09-03', '2020-09-02', '2020-09-01'],
         'random_order': [7, 6, 8, 0, 2, 4, 3, 1, 5],
         'repeating_dates': ['2020-08-01', '2019-08-01', '2020-08-01', '2012-08-01', '2019-08-01', '2019-08-01', '2019-08-01', '2013-08-01', '2019-08-01'],
         'special': [7, 8, 0, 1, 4, 2, 6, 3, 5],
         'special_dates': ['2020-08-01', '2019-08-01', '2020-08-01', '2012-08-01', '2019-08-01', '2019-08-01', '2019-08-01', '2013-08-01', '2019-08-01'],
     })
-    df.ww.init(index='backwards_order', time_index='dates_backwards')
+
+
+def test_add_secondary_time_index(dates_df):
+    dates_df.ww.init(index='backwards_order', time_index='dates_backwards')
     es = EntitySet('es')
-    es.add_dataframe('dates_table', df, secondary_time_index={'repeating_dates': ['random_order', 'special']})
+    es.add_dataframe('dates_table', dates_df, secondary_time_index={'repeating_dates': ['random_order', 'special']})
+
+    assert dates_df.ww.metadata['secondary_time_index'] == {'repeating_dates': ['random_order', 'special', 'repeating_dates']}
+
 
+def test_time_type_check_order(dates_df):
+    dates_df.ww.init(index='backwards_order', time_index='random_order')
+    es = EntitySet('es')
+
+    error = 'dates_table time index is numeric type which differs from other entityset time indexes'
+    with pytest.raises(TypeError, match=error):
+        # Because we set secondary time index before checking that time_index is valid, the time type will match the secondary time index
+        es.add_dataframe('dates_table', dates_df, secondary_time_index={'repeating_dates': ['random_order', 'special']})
+
+    # Metadata on the woodwork table still gets set --> maybe we should reverse the order here
     assert df.ww.metadata['secondary_time_index'] == {'repeating_dates': ['random_order', 'special', 'repeating_dates']}
 
 
+def test_add_time_index_through_woodwork_different_type(dates_df):
+    dates_df.ww.init(index='backwards_order')
+    es = EntitySet('es')
+
+    es.add_dataframe('dates_table', dates_df, secondary_time_index={'repeating_dates': ['random_order', 'special']})
+
+    assert dates_df.ww.metadata['secondary_time_index'] == {'repeating_dates': ['random_order', 'special', 'repeating_dates']}
+    assert es.time_type == Datetime
+
+    assert es._check_uniform_time_index(es['dates_table']) is None
+
+    # --> bug? IF a user sets the time index after initalization through woodwork, there's no way to check the time type
+    # --> should we add a entityset add time index method to help with this??
+    dates_df.ww.set_time_index('random_order')
+    assert dates_df.ww.time_index == 'random_order'
+
+    error = 'dates_table time index is numeric type which differs from other entityset time indexes'
+    with pytest.raises(TypeError, match=error):
+        es._check_uniform_time_index(es['dates_table'])
```

Comment:
If a user wants to change the time index of their DataFrame in an EntitySet, they'll go through woodwork's `set_time_index`, which means that we don't get the time type check, and a user could set a time index that is the incorrect time type, and they won't catch it until later. 

Are we okay with this? If not, we might want to add a `EntitySet.set_dataframe_time_index` method that confirms the time type is valid for the EntitySet.

Reply:
The idea behind PR #1400 was to try to gracefully catch time type issues for the scenario you describe. My thought is to try to keep things simple and not implement an additional method for setting the time index for now, but we could revisit if it proves to be a problem later.

Reply:
ah, okay, I'm seeing that part of the conversation now.

The time type checking on EntitySet does a really good job of catching time type issues, so it sounds like we can leverage it later on in dfs to let users know if there's an issue with mismatched time types.